tools/boot.sh

```diff
@@ -37,7 +37,11 @@ BOOT_BYTES=$(wc -c "${SYNCLOUD_IMAGE}" | cut -f 1 -d ' ')
 BOOT_SECTORS=$(( ${BOOT_BYTES} / 512 ))
 echo "boot sectors: ${BOOT_SECTORS}"
 ROOTFS_SECTORS=$( numfmt --from=iec --to-unit=512 $ROOTFS_SIZE )
-dd if=/dev/zero count=${ROOTFS_SECTORS} >> ${SYNCLOUD_IMAGE}
+COUNT_TO_ROOTFS=${ROOTFS_SECTORS}
+if [[ ${SYNCLOUD_BOARD} == "jetson-nano" ]];then
+  COUNT_TO_ROOTFS=$((${COUNT_TO_ROOTFS} + 2048))
```

Comment:
1. ideally we include board name only in extract stage
2. why do you need additional space in the image here?
We have a way to expand rootfs to take full disk free space later to reduce the image size (download speed).
There are two ways of expanding after you write an image:
UI: Settings - Internal Memory page (does not work, I am fixing right now :))
CLI: `/snap/platform/current/bin/boot_extend.sh`

Reply:
1. I'm not sure if WSL2 caused this problem as logged as below.
2. Log of below shows why I need to incrace the size after passing the rootfs size to make boot stage success.

Here is log of `sudo drone exec --pipeline=jetson-nano-all-buster --include=boot --trusted`
```
[boot:199] + export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
[boot:200] + PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
[boot:201] + export DEBCONF_FRONTEND=noninteractive
[boot:202] + DEBCONF_FRONTEND=noninteractive
[boot:203] + export DEBIAN_FRONTEND=noninteractive
[boot:204] + DEBIAN_FRONTEND=noninteractive
[boot:205] + export TMPDIR=/tmp
[boot:206] + TMPDIR=/tmp
[boot:207] + export TMP=/tmp
[boot:208] + TMP=/tmp
[boot:209] + SRC_ROOTFS=rootfs_jetson-nano
[boot:210] + DST_ROOTFS=dst_jetson-nano/root
[boot:211] + echo 'copying boot'
[boot:212] + cp jetson-nano/boot syncloud-jetson-nano-.img
[boot:213] copying boot
[boot:214] ++ wc -c syncloud-jetson-nano-.img
[boot:215] ++ cut -f 1 -d ' '
[boot:216] + BOOT_BYTES=285263360
[boot:217] + BOOT_SECTORS=557155
[boot:218] boot sectors: 557155
[boot:219] + echo 'boot sectors: 557155'
[boot:220] ++ numfmt --from=iec --to-unit=512 3G
[boot:221] + ROOTFS_SECTORS=6291456
[boot:222] + COUNT_TO_ROOTFS=6291456
[boot:223] + [[ jetson-nano == \j\e\t\s\o\n\-\n\a\n\o ]]
[boot:224] + echo 'It'\''s jetson nano'
[boot:225] + dd if=/dev/zero bs=512 count=6291456
[boot:226] It's jetson nano
[boot:227] 6291456+0 records in
[boot:228] 6291456+0 records out
[boot:229] 3221225472 bytes (3.2 GB, 3.0 GiB) copied, 7.2091 s, 447 MB/s
[boot:230] + ROOTFS_START_SECTOR=557156
[boot:231] + ROOTFS_END_SECTOR=6848512
[boot:232] + fdisk -l syncloud-jetson-nano-.img
[boot:233] GPT PMBR size mismatch (4825087 != 6848610) will be corrected by write.
[boot:234] The backup GPT table is corrupt, but the primary appears OK, so that will be used.
[boot:235] The backup GPT table is not on the end of the device. This problem will be corrected by write.
[boot:236] Disk syncloud-jetson-nano-.img: 3.3 GiB, 3506488832 bytes, 6848611 sectors
[boot:237] Units: sectors of 1 * 512 = 512 bytes
[boot:238] Sector size (logical/physical): 512 bytes / 512 bytes
[boot:239] I/O size (minimum/optimal): 512 bytes / 512 bytes
[boot:240] Disklabel type: gpt
[boot:241] Disk identifier: 2EA8566B-855D-CF41-A5E3-CBEF02518A8B
[boot:242]
[boot:243] Device                     Start    End Sectors  Size Type
[boot:244] syncloud-jetson-nano-.img1 32768 557055  524288  256M EFI System
[boot:245] + PTTYPE=gpt
[boot:246] + echo 'creating second partition (557156 - 6848512) sectors'
[boot:247] creating second partition (557156 - 6848512) sectors
[boot:248] + [[ gpt == \g\p\t ]]
[boot:249] ++ losetup -f --show syncloud-jetson-nano-.img
[boot:250] + LOOP=/dev/loop0
[boot:251] + echo '
[boot:252] r
[boot:253] d
[boot:254] w
[boot:255] Y
[boot:256] Y
[boot:257] '
[boot:258] + gdisk /dev/loop0
[boot:259] Caution: invalid backup GPT header, but valid main header; regenerating
[boot:260] backup header from main header.
[boot:261]
[boot:262] GPT fdisk (gdisk) version 1.0.3
[boot:263]
[boot:264] Warning! Main and backup partition tables differ! Use the 'c' and 'e' options
[boot:265] on the recovery & transformation menu to examine the two tables.
[boot:266]
[boot:267] Warning! One or more CRCs don't match. You should repair the disk!
[boot:268]
[boot:269] Partition table scan:
[boot:270]   MBR: protective
[boot:271]   BSD: not present
[boot:272]   APM: not present
[boot:273]   GPT: damaged
[boot:274]
[boot:275] ****************************************************************************
[boot:276] Caution: Found protective or hybrid MBR and corrupt GPT. Using GPT, but disk
[boot:277] verification and recovery are STRONGLY recommended.
[boot:278] ****************************************************************************
[boot:279]
[boot:280] Command (? for help):
[boot:281] Command (? for help):
[boot:282] Recovery/transformation command (? for help):
[boot:283] Recovery/transformation command (? for help): Warning! Secondary header is placed too early on the disk! Do you want to
[boot:284] correct this problem? (Y/N): Have moved second header and partition table to correct location.
[boot:285]
[boot:286] Final checks complete. About to write GPT data. THIS WILL OVERWRITE EXISTING
[boot:287] PARTITIONS!!
[boot:288]
[boot:289] Do you want to proceed? (Y/N): OK; writing new GUID partition table (GPT) to /dev/loop0.
[boot:290] Warning: The kernel is still using the old partition table.
[boot:291] The new table will be used at the next reboot or after you
[boot:292] run partprobe(8) or kpartx(8)
[boot:293] The operation has completed successfully.
[boot:294] + sgdisk -n 2:557156:6848512 -p /dev/loop0
[boot:295] Could not create partition 2 from 557156 to 6848512
[boot:296] Error encountered; not saving changes.
[boot:297] Information: Moved requested sector from 557156 to 557056 in
[boot:298] order to align on 2048-sector boundaries.
[boot:299] Disk /dev/loop0: 6848611 sectors, 3.3 GiB
[boot:300] Sector size (logical/physical): 512/512 bytes
[boot:301] Disk identifier (GUID): 2EA8566B-855D-CF41-A5E3-CBEF02518A8B
[boot:302] Partition table holds up to 128 entries
[boot:303] Main partition table begins at sector 2 and ends at sector 33
[boot:304] First usable sector is 2048, last usable sector is 6846563
[boot:305] Partitions will be aligned on 2048-sector boundaries
[boot:306] Total free space is 6320228 sectors (3.0 GiB)
[boot:307]
[boot:308] Number  Start (sector)    End (sector)  Size       Code  Name
[boot:309]    1           32768          557055   256.0 MiB   EF00  efi
```

Reply:
Now I'm clear that WSL2 wasn't cause that,because in [CI](http://ci.syncloud.org:8080/syncloud/image/291/3/3) it's happend too.
And I found a new way to increase usable sectors to solve this problem. It's `sgdisk $LOOP -E`, I can fill missing sectors to make `ROOTFS_END_SECTOR` result working or modify the `ROOTFS_END_SECTOR` to the output of `sgdisk $LOOP -E`.

Reply:
I see, 2048 sectors is just one megabyte if it makes easier you can leave this without condition for every board.

Reply:
sgdisk -E is only for reading so you can save it and use the doc says.
Also -n can take 0 which means max for the end.
Like `sgdisk -n 2:557156:0 -p /dev/loop0`

Reply:
Finally I use the `sgdisk -E`'s result to correct the `ROOTFS_END_SECTOR` like
```shell
  USABLE_SECTORS=$(sgdisk $LOOP -E 2>/dev/null)
  if [[ ${ROOTFS_END_SECTOR} -gt ${USABLE_SECTORS} ]]; then
     ROOTFS_END_SECTOR=$USABLE_SECTORS
  fi
```